### ISSUE TITLE
Ordering of species in reaction is nondeterministic

### ISSUE THREAD

**Issue by connie** [ReactionMechanismGenerator/RMG-Py#409]
When a reaction string is printed in Chemkin in different jobs, the ordering in which they are listed gets jumbled.  For instance in 2 jobs, I can get:

```
CH3(17)+CH3(17)=H(2)+C2H5(27)                       7.845e+12 0.100     10.600   
```

and 

```
CH3(17)+CH3(17)=C2H5(27)+H(2)                       7.845e+12 0.100     10.600  
```

Which are identical except for the ordering.  This is correct but makes it difficult for the user when one has to search multiple permutations of a single reaction to identify it.  I think we should try to make str(reaction) print the species in order of index.

**Comment by rwest** [comment]
#388 may be relevant here

**Comment by mliu49** [comment]
Species sorting now uses comparison methods (implemented in 0c0c7ed13c5eac36a2b4d073b91156e448e5c3d7), which use the new `sorting_key` property (implemented in c90c407ad93eb32de7cc5a6c0cd2d8192a4b5145). This will provide deterministic sorting, although the current implementation depends on the species label and index, which can change depending on thermo libraries and the order that species are generated.

In cases like this, where the indices are the same, this will guarantee deterministic results.
